dlib/control/approximate_linear_models_abstract.h

```diff
@@ -56,41 +59,29 @@ namespace dlib
                 - returns the dimensionality of the PSI() feature vector.  
         !*/
 
-        action_type find_best_action (
-            const state_type& state,
-            const matrix<double,0,1>& w
-        ) const;
-        /*!
-            ensures
-                - returns the action A that maximizes Q(state,A) = dot(w,PSI(state,A)).
-                  That is, this function finds the best action to take in the given state
-                  when our model is parameterized by the given weight vector w.
-        !*/
-
-        void get_features (
-            const state_type& state,
-            const action_type& action,
-            matrix<double,0,1>& feats
+        matrix<double,0,1> get_features (
```

Comment:
This was changed in the new code, but the LSPI code still does it the old way.  Everything needs to be the same.

Also, don't do this.  This change makes the code slower, since now each call to get_features allocates a block of memory.  The other way avoids the reallocation.  I realize move constructors make returning the matrix not super slow.  But reallocating memory in a tight loop is also bad.